### PR TITLE
ML-046/add comments

### PR DESCRIPTION
**Pull request by WhatWasThatt**
# Description

add comments model
add commentable polymorphic association
add comment controller
add routes for comments
add can leave comment permission
add comment policy
add comment factory
add comment model specs
add comment policy specs
add comment request specs
add comment routing specs

## Trello card link

https://trello.com/c/3JE

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [x] Does your code require any migrations to be ran?
- [ ] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

**Review by twill11 — DISMISSED** [review]
Look's perfect, good job!

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+module Api
+  module V1
+    class CommentsController < ApplicationController
+      before_action :set_comment, only: %i[update destroy]
+
+      def create
+        @comment = @commentable.comments.build(comment_params)
+        @comment.user = current_user
```
Comment (ShrewHools) [inline app/controllers/api/v1/comments_controller.rb:8]: What happens if I am not logged in to the system?

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+module Api
+  module V1
+    class CommentsController < ApplicationController
+      before_action :set_comment, only: %i[update destroy]
+
+      def create
+        @comment = @commentable.comments.build(comment_params)
```
Comment (ShrewHools) [inline app/controllers/api/v1/comments_controller.rb:7]: @commentable nil?)

File: app/controllers/api/v1/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      end
+
+      def destroy
+        @comment.destroy
+      end
+
+      private
+
+      def set_comment
+        @company = @commentable.comments.find(params[:id])
```
Comment (ShrewHools) [inline app/controllers/api/v1/comments_controller.rb:32]: commentable nil?

File: spec/requests/api/v1/comments_spec.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  let(:movie_comment) { create(:comment) }
+  let(:person_comment) { create(:comment, :for_person) }
+  let(:nested_comment) { create(:comment, :for_comment) }
+
+  describe 'POST /create' do
+    context 'with valid parameters' do
+      it 'creates a new movie Comment' do
+        expect do
+          post api_v1_movie_comments_url(movie_id: movie_comment.commentable_id),
```
Comment (ShrewHools) [inline spec/requests/api/v1/comments_spec.rb:24]: what happens when "movie_comment.commentable_id" is called?
sometimes you can check the functionality by hand)))

Comment (WhatWasThatt) [reply spec/requests/api/v1/comments_spec.rb:24]: image
Can you specify?

File: app/controllers/api/v1/comments_controller.rb (hunk 1 above)
Comment (WhatWasThatt) [reply app/controllers/api/v1/comments_controller.rb:8]: Later, an unregistered user will not have access to these actions

File: app/controllers/api/v1/comments_controller.rb (hunk 2 above)
Comment (WhatWasThatt) [reply app/controllers/api/v1/comments_controller.rb:7]: ok, i'll fix it)

**Review by twill11 — DISMISSED** [review]
Good job!

**Review by KarinaKlevzhits — APPROVED** [review]

**Review by dmitrymalllpr — APPROVED** [review]

**Pull request merged**